core/embed/rust/src/ui/model_mercury/component/share_words.rs

```diff
@@ -18,11 +21,117 @@ use heapless::{String, Vec};
 
 const MAX_WORDS: usize = 33; // super-shamir has 33 words, all other have less
 const ANIMATION_DURATION_MS: Duration = Duration::from_millis(166);
+type IndexVec = Vec<usize, MAX_WORDS>;
 
 /// Component showing mnemonic/share words during backup procedure. Model T3T1
 /// contains one word per screen. A user is instructed to swipe up/down to see
 /// next/previous word.
+/// This is a wrapper around a Frame so that the subtitle and Footer of the
+/// Frame can be updated based on the index of the word shown. Actual share
+/// words are rendered within `ShareWordsInner` component,
 pub struct ShareWords<'a> {
+    area: Rect,
```

Comment:
Doesn't seem to be used anywhere, same for ShareWordsInner.

Reply:
Thanks, removed everywhere https://github.com/trezor/trezor-firmware/pull/3954/commits/856b4578701228e136642e379a8e00c2de9aa2a7